src/main/java/net/fabricmc/lorenztiny/TinyMappingsReader.java

```diff
@@ -72,22 +72,35 @@ public TinyMappingsReader(final MappingTree tree, final String from, final Strin
 	@Override
 	public MappingSet read(final MappingSet mappings) {
 		for (final MappingTree.ClassMapping klass : this.tree.getClasses()) {
-			final ClassMapping<?, ?> mapping = mappings.getOrCreateClassMapping(klass.getName(this.from))
-					.setDeobfuscatedName(klass.getName(this.to));
+			final String classNameTo = klass.getName(this.to);
+			final String classNameFrom = klass.getName(this.from);
+			if (classNameTo == null || classNameFrom == null) {
+				continue;
+			}
+			final ClassMapping<?, ?> mapping = mappings.getOrCreateClassMapping(classNameFrom)
+					.setDeobfuscatedName(classNameTo);
 
 			for (final MappingTree.FieldMapping field : klass.getFields()) {
-				mapping.getOrCreateFieldMapping(field.getName(this.from), field.getDesc(this.from))
-						.setDeobfuscatedName(field.getName(this.to));
+				final String fieldNameTo = field.getName(this.to);
+				final String fieldNameFrom = field.getName(this.from);
```

Comment:
I believe these "from" names can be null too? With custom namespaces they are not necessarily in the mandatory populated NS column.